pygmt/clib/session.py

```diff
@@ -1475,7 +1475,7 @@ def virtualfile_from_vectors(
         # 2 columns contains coordinates like longitude, latitude, or datetime string
         # types.
         for col, array in enumerate(arrays[2:]):
-            if pd.api.types.is_string_dtype(array.dtype):
+            if np.issubdtype(array.dtype, np.str_):
```

Comment:
Changes in this PR were initially added in #684 to support string arrays with np.object_ dtype. It's no longer necessary after dac7e8eda7632464f556eaae5f1ac8ac34c9b6bd.